pupil_src/shared_modules/gaze_producers.py

```diff
@@ -331,17 +327,43 @@ def remove():
 
             return remove
 
+        def set_trim_fn(button, sec, key):
+            def trim(format_only=False):
+                if format_only:
+                    left_idx, right_idx = sec[key]
+                else:
+                    right_idx = self.g_pool.seek_control.trim_right
+                    left_idx = self.g_pool.seek_control.trim_left
+                    sec[key] = left_idx, right_idx
+
+                time_fmt = key.replace('_', ' ').title() + ': '
+                min_ts = self.g_pool.timestamps[0]
+                for idx in (left_idx, right_idx):
+                    ts = self.g_pool.timestamps[idx] - min_ts
+                    minutes = ts // 60
+                    seconds = ts - (minutes * 60.)
+                    time_fmt += ' {:02.0f}:{:02.0f} -'.format(abs(minutes), seconds)
+                button.outer_label = time_fmt[:-2]  # remove final ' - '
+            button.function = trim
+
         section_menu.append(ui.Text_Input('label', sec, label='Label'))
         section_menu.append(ui.Selector('calibration_method', sec,
                                         label="Calibration Method",
                                         labels=['Circle Marker', 'Natural Features'],
                                         selection=['circle_marker', 'natural_features']))
-        section_menu.append(ui.Selector('mapping_method', sec, label='Calibration Mode',selection=['2d','3d']))
+        section_menu.append(ui.Selector('mapping_method', sec, label='Calibration Mode',selection=['2d', '3d']))
         section_menu.append(ui.Text_Input('status', sec, label='Calibration Status', setter=lambda _: _))
-        section_menu.append(ui.Text_Input('calibration_range', sec, label='Calibration range',
-                                          setter=make_validate_fn(sec, 'calibration_range')))
-        section_menu.append(ui.Text_Input('mapping_range', sec, label='Mapping range',
-                                          setter=make_validate_fn(sec, 'mapping_range')))
+
+        calib_range_button = ui.Button('Trim calibration range', None)
```

Comment:
Info Text:
This section is calibrated using reference markers found in a user set range "Calibration". The calibration is used to map pupil to gaze positions within a user set range "Mapping". Drag trim marks in the timeline to set a range and apply it.

Label: 
Calibration 02:00 - 02:56 [Set from trim marks]
Mapping 00:00 - 10:00 [Set from trim marks]